client/my-sites/term-selector/index.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import TermSelectorTerms from './terms';
+
+export default React.createClass( {
+	propTypes: {
+		multiple: PropTypes.bool,
+		className: PropTypes.string,
+		onChange: PropTypes.func.isRequired,
+		selected: PropTypes.array,
+		createLink: PropTypes.string,
+		analyticsPrefix: PropTypes.string,
+		taxonomy: PropTypes.string
+	},
+
+	getDefaultProps() {
+		return {
+			analyticsPrefix: 'Category Selector',
+			selected: [],
+			taxonomy: 'category',
+			onChange: () => {}
+		};
+	},
+
+	getInitialState() {
+		return {
+			page: 1,
+			search: ''
+		};
+	},
+
+	onSearch( searchTerm ) {
+		if ( searchTerm !== this.state.search ) {
+			this.setState( {
+				page: 1,
+				search: searchTerm
+			} );
+		}
+	},
+
+	incrementPage() {
+		this.setState( {
+			page: this.state.page + 1
+		} );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		const isChangingQuery = [
+			'taxonomy',
+			'multiple'
```

Comment:
In which case, this entire function could be simplified to a single `nextProps.taxonomy !== this.props.taxonomy` condition.